katdal/concatdata.py

```diff
@@ -341,10 +417,11 @@ def get(self, name, select=False, extract=True, **kwargs):
 
         """
         # Get array, categorical data or raw sensor data from each cache
-        split_data = [cache.get(name, select, extract, **kwargs) for cache in self.caches]
-        # If this sensor has already been partially extracted, we are forced to extract it in rest of caches too
-        if not extract and not np.all([isinstance(data, SensorData) for data in split_data]):
-            split_data = [cache.get(name, select, True, **kwargs) for cache in self.caches]
+        split_data = self._get(name, select, extract, **kwargs)
+        # If this sensor has already been partially extracted,
+        # we are forced to extract it in rest of caches too
+        if not extract and not all(isinstance(sd, SensorData) for sd in split_data):
+            split_data = self._get(name, select, True, **kwargs)
```

Comment:
In what cases would `isinstance(sd, SensorData)` be false? Did you mean `type(sd) is SensorData`?